src/bots/DepositBot.js

```diff
@@ -0,0 +1,272 @@
+const loggerNamespace = 'dx-service:bots:DepositBot'
+const Bot = require('./Bot')
+const Logger = require('../helpers/Logger')
+const getVersion = require('../helpers/getVersion')
+
+const getBotAddress = require('../helpers/getBotAddress')
+
+const numberUtil = require('../../src/helpers/numberUtil')
+const dateUtil = require('../../src/helpers/dateUtil')
+
+const logger = new Logger(loggerNamespace)
+// const auctionLogger = new AuctionLogger(loggerNamespace)
+
+const ETHER_RESERVE_AMOUNT =
+  process.env.ETHER_RESERVE_AMOUNT || 1.5
+const DEPOSIT_PERIODIC_CHECK_MILLISECONDS =
+  process.env.DEPOSIT_BOT_CHECK_TIME_MS || (5 * 60 * 1000) // 5 min
+
+class DepositBot extends Bot {
+  constructor ({
+    name,
+    dxInfoService,
+    dxTradeService,
+    ethereumClient,
+    slackClient,
+    botTransactionsSlackChannel,
+    tokensByAccount,
+    notifications,
+    checkTimeInMilliseconds = DEPOSIT_PERIODIC_CHECK_MILLISECONDS,
+    inactiveTimeLapse = []
+  }) {
+    super(name)
+    this._dxInfoService = dxInfoService
+    this._dxTradeService = dxTradeService
+    this._ethereumClient = ethereumClient
+    this._slackClient = slackClient
+    this._botTransactionsSlackChannel = botTransactionsSlackChannel
+
+    this._tokensByAccount = tokensByAccount
+
+    this._notifications = notifications
+    this._checkTimeInMilliseconds = checkTimeInMilliseconds
+
+    this._inactiveTimeLapse = inactiveTimeLapse
+
+    this._lastCheck = null
+    this._lastDeposit = null
+    this._lastError = null
+
+    this._botInfo = 'DepositBot - v' + getVersion()
+  }
+
+  async _doStart () {
+    logger.debug({ msg: 'Initialized bot: ' + this.name })
+
+    // Check if the bots need to deposit periodically
+    this._depositFunds()
+    setInterval(() => {
+      return this._depositFunds()
+    }, this._checkTimeInMilliseconds)
+  }
+
+  async _doStop () {
+    logger.debug({ msg: 'Bot stopped: ' + this.name })
+  }
+
+  async _depositFunds () {
+    this._lastCheck = new Date()
+    let botHasDepositedFunds
+
+    function isInactiveTimeLapse ({ from, to }) {
```

Comment:
Why this is a inner function? It doesn't have state or doesn't use the clousures.

I would move it out, or it can be even be left in dateUtil: i.e. `isNowInTimeWindow" or sth like that.

The implementation using some, very nice :)